ui/app/[locale]/(authorized)/kafka/[kafkaId]/nodes/NodesTable.tsx

```diff
@@ -1,259 +1,419 @@
+    ControllerStatusLabel,
+  ) as ControllerStatus[];
+
   return (
-    <ResponsiveTable
-      ariaLabel={"Kafka nodes"}
-      columns={columns}
-      data={nodes}
-      renderHeader={({ column, key, Th }) => {
-        switch (column) {
-          case "id":
-            return <Th key={key}>{t("nodes.broker_id")}</Th>;
-          case "roles":
-            return <Th key={key}>{t("nodes.roles")}</Th>;
-          case "status":
-            return <Th key={key}>{t("nodes.status")}</Th>;
-          case "replicas":
-            return (
-              <Th key={key}>
-                {t("nodes.replicas")}{" "}
-                <Tooltip
-                  content={
-                    t("nodes.replicas_tooltip")
-                  }
-                >
-                  <HelpIcon />
-                </Tooltip>
-              </Th>
-            );
-          case "rack":
+    <Card>
+      <CardHeader>
+        <CardTitle className="pf-v5-u-pl-md">{t("nodes.title")}</CardTitle>
+      </CardHeader>
+      <CardBody>
+        <TableView
+          page={page}
+          perPage={perPage}
+          onPageChange={onPageChange}
+          itemCount={nodesCount}
+          isColumnSortable={isColumnSortable}
+          isRowExpandable={() => true}
+          onClearAllFilters={onClearAllFilters}
+          data={nodeList}
+          emptyStateNoData={<></>}
+          emptyStateNoResults={
+            <EmptyStateNoMatchFound onClear={onClearAllFilters!} />
+          }
+          ariaLabel={"Kafka nodes"}
+          columns={NodeListColumns}
+          renderHeader={({ column, key, Th }) => {
+            switch (column) {
+              case "id":
+                return <Th key={key}>{t("nodes.broker_id")}</Th>;
+              case "roles":
+                return <Th key={key}>{t("nodes.roles")}</Th>;
+              case "status":
+                return <Th key={key}>{t("nodes.status")}</Th>;
+              case "replicas":
+                return (
+                  <Th key={key} width={20}>
+                    {t("nodes.replicas")}{" "}
+                    <Tooltip content={t("nodes.replicas_tooltip")}>
+                      <HelpIcon />
+                    </Tooltip>
+                  </Th>
+                );
+              case "rack":
+                return (
+                  <Th key={key}>
+                    {t("nodes.rack")}{" "}
+                    <Tooltip content={t("nodes.rack_tooltip")}>
+                      <HelpIcon />
+                    </Tooltip>
+                  </Th>
+                );
+              case "nodePool":
+                return <Th key={key}>{t("nodes.nodePool")}</Th>;
+            }
+          }}
+          renderCell={({ column, key, row, Td }) => {
+            switch (column) {
+              case "id":
+                return (
+                  <Td key={key} dataLabel={"Node ID"}>
+                    <Link href={`nodes/${row.id}`}>{row.id}</Link>
```

Comment:
Controller-only node configuration not retrievable - make link available only when the broker role is present. Maybe we need a tool-tip to indicate the controller configuration is not available?

```suggestion
                    {row.attributes.broker
                        ? <Link href={`nodes/${row.id}`}>{row.id}</Link>
                        : <>{row.id}</>
                    }
```